Generate.py

```diff
@@ -378,6 +378,46 @@ def roll_linked_options(weights: dict) -> dict:
     return weights
 
 
+def handle_trigger_math(math_options: Union[dict, string], option_name: str, weights: dict):
+    options = copy.deepcopy(math_options)
+    if not isinstance(options, dict):
+        options = {str(options): 1}
+    new_dict = {}
+    for name, value in options.items():
+        value_list = []
+        if not isinstance(name, str):
+            raise Exception(f'Mathematical trigger malformed: {name} is not a string.')
+        split_name = name.split(" ")
+        if len(split_name) != 3:
+            raise Exception(f'Cannot perform arithmetic, wrong number of arguments: {math_options}')
+        if split_name[1] not in ['+', '-', '*', '/']:
+            raise Exception(f'Cannot perform arithmetic, unknown operator in option: {split_name[1]}')
+        for x in range(0, 2, 2):
+            if not split_name[x].isnumeric():
+                if split_name[x] in weights:
+                    weights[split_name[x]] = get_choice(split_name[x], weights)
+                    split_name[x] = str(weights[split_name[x]])
+                if not split_name[x].isnumeric():
+                    raise Exception(f'Cannot perform arithmetic, non-numeric value found for : {name.split(" ")[0]}')
```

Comment:
Honestly, you can probably remove the colon entirely, right?
```suggestion
                    raise Exception(f'Cannot perform arithmetic, non-numeric value found for: {name.split(" ")[0]}')
```